nginx-controller/nginx/configurator.go

```diff
@@ -587,10 +704,14 @@ func GenerateCertAndKeyFileContent(secret *api_v1.Secret) []byte {
 func (cnf *Configurator) DeleteSecret(key string, ings []extensions.Ingress) error {
 	for _, ing := range ings {
 		cnf.nginx.DeleteIngress(objectMetaToFileName(&ing.ObjectMeta))
+
+		delete(cnf.ingresses, fmt.Sprintf("%s/%s", ing.Namespace, ing.Name))
 	}
 
 	cnf.nginx.DeleteSecretFile(keyToFileName(key))
 
+	cnf.updateMaps()
```

Comment:
Updating maps only make sense when `len(ings) > 0`

Reply:
That condition is already present inside method, not need to duplicate it.

Reply:
This is slightly different.  

DeleteSecret method handles an event, when a secret is removed from k8s. If that secret is referenced in some Ingress resources, those Ingress resources will be temporary removed as well and they are passed in `ings` slice.  So, if that slice is empty, we don't need to regenerate the maps, as there were no changes to Ingress resources.